src/aspire/operators/blk_diag_matrix.py

```diff
@@ -185,9 +200,24 @@ def __check_compatible(self, other):
                 "Number of blocks {} {} are not equal.".format(len(self), len(other))
             )
 
-        self.__check_size_compatible(other)
+        if size_compat == "add":
+            self.__check_size_compatible_add(other)
+        elif size_compat == "mul":
+            self.__check_size_compatible_mul(other)
+        else:
+            raise RuntimeError("Unknown compatibility type {}".format(size_compat))
+
         self.__check_dtype_compatible(other)
 
+    @property
+    def isregular(self):
```

Comment:
Yeah, no problem. Changed to `_issquare`.  Not sure what you mean about convention. I imagined this was something that a user might want to call. Easy to change later if comes up.

Reply:
Forgot to discuss this at our meeting but by convention I just meant naming. For most other methods, this would be `is_square`, `is_finite`, etc. I suspect this naming comes from `np.isfinite`, etc., but it might be more consistent with the rest of the codebase to write with underscores.

Reply:
I'll leave it up to you which you think would work best.

Reply:
oh okay, gotcha. Yes, that makes sense. I'll change it to resolve this discussion, then merge. Thanks.